modules/um_callerid.c

```diff
@@ -248,8 +249,32 @@ h_hdl_privmsg_user(void *vdata)
 		return;
 	}
 
+	/* Check if +M is being used to override +g */
+	bool would_block = false;
+	if (MyClient(target_p) && IsSetAnyCallerID(target_p) && IsPerson(source_p))
+	{
+		if (!(IsSetRelaxedCallerID(target_p) &&
+				!IsSetStrictCallerID(target_p) &&
+				has_common_channel(source_p, target_p)) &&
+			!accept_message(source_p, target_p) &&
+			!MayHavePrivilege(source_p, "oper:always_message"))
+		{
+			would_block = true;
+		}
```

Comment:
This duplicates the logic in `allow_message`, which is both inefficient and makes it more likely that future changes will not be applied correctly to both copies. Please move this determination to `allow_message`, making it return one out of three possibilities.